File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        int start=0,maxlen=0;
        for(int i=0;i<s.size();++i){
            for(int j=0;j<2;++j){
                int l=i,r=i+j;
                while(l>=0 && r<s.size()&& s[l]==s[r]){
                    if(r-l+1 > maxlen){
                        start = l,maxlen=r-l+1;
                        
                    }
                    --l;
                    ++r;
                }
            }
        }
        return s.substr(start,maxlen);
    }
};

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """
        def expand(l,r):
            while l>=0 and r<len(s) and s[l]==s[r]:
                l-=1
                r+=1
            return s[l+1:r]
        res=""
        for i in range(len(s)):
            temp=expand(i,i)
            if len(temp)>len(res):res=temp
            temp=expand(i,i+1)
            if len(temp)>len(res):res=temp
        return res